### DISCUSSION TITLE
Deep potential of later iterations give worse predictions than previous iterations

### DISCUSSION THREAD

**Discussion by PolyuWeldingSpock** [deepmodeling/dpgen#1365] [Q&A]
Hi DPGEN developers,

Recently I have tried to develop a deep potential which could apply on cubic and tetragonal phase of CsPbI3. Initial dataset was got from init_bulk method. After 4 iterations, I extracted the deep potential to compare the DP with DFT calculated equation of states of cubic phase of CsPbI3. 
![f02d325568906c03599e8c3a8b3370c9](https://github.com/deepmodeling/dpgen/assets/122458488/b1ebf5f5-9540-4327-9c04-cde25584edc6)
Then I run 1 more iteration and did the same thing. 
![6d5e571663b809a8d5f111ad6c00d5a2](https://github.com/deepmodeling/dpgen/assets/122458488/61496ed6-1e35-44ac-9949-1f79966d8d93)
However, even two DP results both did not showed correct trend, results of  iteration 5 is much worse (deviation is greater than iter 4). I want to know that should be later deep potential is always better than previous ? or My case is just abnormal due to some reason?

**Comment by njzjz** [comment]
I suggest checking the new data in the new iteration which may give unreasonable energy. Also, check whether `sel` is still enough.

**Reply by PolyuWeldingSpock** [reply]
Thanks for your reply @njzjz , I have set `sel` to "auto" in all my iterations. The train.log showed that actual `sel` for Cs Pb and I are 13 12 36 
![59b07c756c5058a0d68fcbfdb200eeb3](https://github.com/deepmodeling/dpgen/assets/122458488/31d34ca5-0d84-44d1-a9ca-82eec5d16c41)
I want to know how to check whether this number is big enough for my elements?

**Reply by njzjz** [reply]
If `auto` is used, you should not worry about it.

**Reply by njzjz** [reply]
> the new data in the new iteration which may give unreasonable energy.

This is more possible

**Reply by PolyuWeldingSpock** [reply]
@njzjz Thanks for your advice. I have check the energy.raw in two iterations. In iteration 4 and previous iterations, most energy values are around -120 eV, but in iteration 5 energy values are around -115 eV.  I thought this may be the cause. However, I am wondering the reason.
First I checked the convergence and confirmed that settings in VASP (KSPACING, ENCUT) were converged. Then I found that most candidate configurations are from high temperature exploration (900K, 1000K and 1 & 10 GPa). I thought there might be some solutions to my situation:
(1) DO not explore these high temperatures in this iteration and later iterations.
(2) Lower the up and low limit of f_deviation when exploring high temperature
(3) Just continue training.  The results might can be enhanced when the dataset contains more configurations from high temperature.
Which solution you think is better?

**Reply by njzjz** [reply]
It makes sense to me. The model does not have enough energy training, so it is close to the initial energy bias (initiated from data). Increasing the prefactor of the energy loss and increasing the number of steps will help.

When performing the active learning cycles, you can consider either strategy: carefully training a model with long steps after all data is collected (considering we do not use the energy for active learning) or in each interaction, starting from the last iteration instead of from scratch.
